android-components/components/feature/search/src/main/java/mozilla/components/feature/search/internal/SearchUrlBuilder.kt

```diff
@@ -44,15 +46,21 @@ internal class SearchUrlBuilder(
 
     private fun buildUrl(template: String, searchTerms: String): String {
         val templateUri = Uri.decode(template)
-        val urlWithSubstitutions = paramSubstitution(templateUri, Uri.encode(searchTerms))
+        val inputEncoding = searchEngine.inputEncoding ?: "UTF-8"
+        val query = try {
+            URLEncoder.encode(searchTerms, inputEncoding).replace("+", "%20")
```

Comment:
To get more context, where does this replacement come from?

Reply:
I had a similar thought and checked out the docs for `URLEncoder` - it has a good description for this :)